tweag/nickel
# Python bindings

This PR adds very simple Python bindings. Fixes https://github.com/tweag/nickel/issues/820.

It builds a package `nickel-lang` with a module `nickel_lang` containing a single function `run` which takes in a `str` of a Nickel expression and returns the result as `str`.

The Python module is built using [maturin](https://github.com/PyO3/maturin).
Note that building the Python modules part is Rust is optional, the binary shouldn't be bigger.

To test this PR:
- go to `pyckel` folder
- create a Python virtual environment and activate it
  ```shell
  python -m venv .venv
  source .venv/bin/activate
  ```
- install maturin
  ```shell
  pip install "maturin>=0.14,<0.15"
  ```
- build and install the `nickel_lang` Python module to the local venv
  ```shell
  maturin develop
  ```
- test with Python
  ```shell
  $ python
  >>> import pyckel
  >>> pyckel.run("let x = 1 in x + 3")
  '4'
  ```

This PR does not
- implement proper Python exceptions: all Nickel errors are just dumped as their Debug formatted string to a NickelException

Todo:
- [x] `run` should return JSON
- [x] `run` should evaluate fully
